pysmt/formula.py

```diff
@@ -231,74 +212,77 @@ def Times(self, left, right):
           - Left and Right must be both INT or REAL type
           - Only linear expressions are allowed
         """
-        n = self.create_node(node_type=op.TIMES, args=(left, right))
-        return n
+        return self.create_node(node_type=op.TIMES, args=(left, right))
 
+    def Pow(self, base, exponent):
+        """ Creates the n-th power of the base.
 
-    def Div(self, left, right):
-        """ Creates an expression of the form:
-            left / right
-
-        Restriction:
-          - Left and Right must be both REAL type
-          - Right is a constant
+        The exponent must be a constant.
         """
-        if not right.is_constant(types.REAL):
-            raise NonLinearError
-        inverse = Fraction(1) / right.constant_value()
-        return self.Times(left, self.Real(inverse))
+        if not exponent.is_constant():
+            raise ValueError("The exponent of POW must be a constant.", exponent)
 
+        if base.is_constant():
+            val = base.constant_value() ** exponent.constant_value()
```

Comment:
Why performing this simplification here? In other operators (e.g. Times or Plus) this is not done and is demanded to the simplifier....